clients/fossid-webapp/src/test/kotlin/FossIdClientTestNewProject.kt

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (C) 2020 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import com.fasterxml.jackson.databind.PropertyNamingStrategies
+import com.fasterxml.jackson.databind.json.JsonMapper
+import com.fasterxml.jackson.module.kotlin.registerKotlinModule
+
+import com.github.tomakehurst.wiremock.WireMockServer
+import com.github.tomakehurst.wiremock.client.WireMock
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration
+
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.collections.shouldBeEmpty
+import io.kotest.matchers.maps.shouldContain
+import io.kotest.matchers.nulls.shouldBeNull
+import io.kotest.matchers.should
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.string.shouldContain
+
+import okhttp3.OkHttpClient
+
+import org.ossreviewtoolkit.fossid.FossIdRestService
+import org.ossreviewtoolkit.fossid.api.Scan
+import org.ossreviewtoolkit.fossid.api.identification.ignored.IgnoredFile
+import org.ossreviewtoolkit.fossid.api.identification.markedAsIdentified.MarkedAsIdentifiedFile
+import org.ossreviewtoolkit.fossid.api.status.DownloadStatus
+import org.ossreviewtoolkit.fossid.api.status.ScanState
+import org.ossreviewtoolkit.fossid.checkDownloadStatus
+import org.ossreviewtoolkit.fossid.checkResponse
+import org.ossreviewtoolkit.fossid.checkScanStatus
+import org.ossreviewtoolkit.fossid.createProject
+import org.ossreviewtoolkit.fossid.createScan
+import org.ossreviewtoolkit.fossid.downloadFromGit
+import org.ossreviewtoolkit.fossid.getProject
+import org.ossreviewtoolkit.fossid.listIdentifiedFiles
+import org.ossreviewtoolkit.fossid.listIgnoredFiles
+import org.ossreviewtoolkit.fossid.listMarkedAsIdentifiedFiles
+import org.ossreviewtoolkit.fossid.listScanResults
+import org.ossreviewtoolkit.fossid.listScansForProject
+import org.ossreviewtoolkit.fossid.runScan
+import org.ossreviewtoolkit.fossid.toList
+import org.ossreviewtoolkit.utils.test.shouldNotBeNull
+
+import retrofit2.Retrofit
+import retrofit2.converter.jackson.JacksonConverterFactory
+
+/**
+ * This client test creates a new project, triggers the download and the scan and gets the scan results
+ */
+class FossIdClientTestNewProject : StringSpec({
```

Comment:
Just wondering because of the name of the test class. Does it only test the creation of a new project? It seems to test broader functionality of the FossId REST client.

Also: Is the test class picked up by Gradle during test execution (I am not sure what naming pattern Gradle expects)?

Reply:
> I am not sure what naming pattern Gradle expects

There should be no naming pattern required; Kotest should pick up any class in a test source set that derives from one of its `Spec` classes.

Reply:
> Just wondering because of the name of the test class. Does it only test the creation of a new project? It seems to test broader functionality of the FossId REST client.

I have a second one in the pipes that run several tests with an existing project.
It is named `FossIdClientTestExistingProject.kt` .
If you find the names unhappy, I am open to suggestions.

Reply:
As usually test classes *end* with `Test`, how about `FossIdClientNewProjectTest`?